lib/metadata/util/win32/Win32Software.rb

```diff
@@ -80,160 +93,150 @@ def initialize(_c, fs)
+      reg_doc
+    end
 
+    def registry_applications(registry_doc)
       # Get the applications
-      reg_node = MIQRexml.findRegElement("HKEY_LOCAL_MACHINE\\SOFTWARE\\Microsoft\\Windows\\CurrentVersion\\Installer\\UserData", reg_doc.root)
-      if reg_node
-        reg_node.each_element_with_attribute(:keyname) do |users|
-          users.each_element_with_attribute(:keyname) do |components|
-            if components.attributes[:keyname].downcase == "products"
-              components.each_element_with_attribute(:keyname) do |products|
-                attrs = XmlFind.decode(products, PRODUCTS_MAPPING)
-                attrs.merge!(:typename => "win32_product", :product_key => @product_keys[attrs[:name]])
-                clean_up_path(attrs)
-                @applications << attrs
-              end
-            end
-          end
+      registry_applications_user_data(registry_doc)
+      ["HKEY_LOCAL_MACHINE\\SOFTWARE\\Wow6432Node\\Microsoft\\Windows\\CurrentVersion\\App Paths",
+       "HKEY_LOCAL_MACHINE\\SOFTWARE\\Microsoft\\Windows\\CurrentVersion\\App Paths"].each do |reg_path|
+        registry_applications_app_paths(registry_doc, reg_path)
+      end
+      ["HKEY_LOCAL_MACHINE\\SOFTWARE\\Wow6432Node\\Microsoft\\Windows\\CurrentVersion\\Uninstall",
+       "HKEY_LOCAL_MACHINE\\SOFTWARE\\Microsoft\\Windows\\CurrentVersion\\Uninstall"].each do |reg_path|
+        reg_node = MIQRexml.findRegElement(reg_path, registry_doc.root)
+        reg_node&.each_element_with_attribute(:keyname) do |e|
+          registry_applications_uninstall(e)
         end
       end
+    end
 
-      ["HKEY_LOCAL_MACHINE\\SOFTWARE\\Wow6432Node\\Microsoft\\Windows\\CurrentVersion\\App Paths",
-       "HKEY_LOCAL_MACHINE\\SOFTWARE\\Microsoft\\Windows\\CurrentVersion\\App Paths"].each do |reg_path|
-        reg_node = MIQRexml.findRegElement(reg_path, reg_doc.root)
-        if reg_node
-          postProcessApps(reg_node, fs)
-          reg_node.each_element_with_attribute(:keyname) do |e|
-            attrs = XmlFind.decode(e, APP_PATHS_MAPPING)
-            next if attrs[:name].nil?
-            attrs.merge!(:typename => "app_path", :product_key => @product_keys[attrs[:name]])
+    def registry_applications_user_data(registry_doc)
+      reg_node = MIQRexml.findRegElement("HKEY_LOCAL_MACHINE\\SOFTWARE\\Microsoft\\Windows\\CurrentVersion\\Installer\\UserData", registry_doc.root)
+      reg_node&.each_element_with_attribute(:keyname) do |users|
+        users.each_element_with_attribute(:keyname) do |components|
+          next unless components.attributes[:keyname].downcase == "products"
+          components.each_element_with_attribute(:keyname) do |products|
+            attrs = XmlFind.decode(products, PRODUCTS_MAPPING)
+            attrs[:typename] = "win32_product"; attrs[:product_key] = @product_keys[attrs[:name]]
             clean_up_path(attrs)
-            @applications << attrs unless isDupApp?(attrs)
```

Comment:
@jerryk55 Why is the `unless isDupApp?(attrs)` no longer needed?

Reply:
Good catch.  That got deleted accidentally.

Reply:
Good catch.  Accidental deletion.